### ISSUE TITLE
Implement LUA binding allowing sending dialogs to clients

### ISSUE THREAD

**Issue by nemerle** [Segs/Segs#146]
Since I've tackled the server->client command queuing we now have ability to send free-form dialogs to client:
![cohdialog](https://user-images.githubusercontent.com/96597/32622850-8fc8ce0a-c584-11e7-8a58-41e930f5e158.png)
Now it would be great to wrap it in LUA:

```lua
sendStandardDialog(client,text)
```

**Comment by broxen** [comment]
Yay!

This is great! On first log in, we can use this to show the WASD help dialog seen below, or some kind of server specific MotD.
![screenshot from 2017-11-10 01-54-32](https://user-images.githubusercontent.com/10159505/32652768-436e8622-c5ba-11e7-95b7-0710518b4312.png)

This could even be handled by a LUA script, for customization, like so in PsuedoLUA:
```lua
WASD_WELCOME = "Welcome to City of Heroes!\nThis short tutorial will go over the basics of gameplay in order to get you started on your path to becoming one of the greatest heroes in Paragon City.\n\nTo get started, walk up to Officer Flint and talk to him. Talking is as simple as left clicking on the person you would like to talk to.\n\nTo move, simply use the keyboard as illustrated here.\n\nUI Wasd.jpg\n\nHolding the right mouse button down and moving the mouse left and right allows you to turn. When you have familiarized yourself with the controls, click OK in the bottom of this dialog box to close it."

function OnFirstSpawn(client)
    sendStandardDialog(client,WASD_WELCOME);
end
```

**Comment by nemerle** [comment]
The preliminary work on coh markup description is here https://github.com/Segs/Segs/wiki/COH-Markup-Language

**Comment by mobbyg** [comment]
Can you provide some examples so I could get a quick feel on how it should work, then I could go explore and hack on my own and perhaps add more to these docs? 

Does it use open and close tags?

<color>white</color> ...etc?

Also does the tag take both arguments in the braces? 

<color>{“color” , ”white”}</color> 

...inside the tags? 

Is it possible to add small images yet like in the example above, and is there a format and size limit?

**Comment by nemerle** [comment]
the code would look like:
```html
<text> Hello <b>Bold</b> hero!</text>
```

**Comment by nemerle** [comment]
Ass for images, they must reference texture name from the client side textures

**Comment by mobbyg** [comment]
So one could add a file and offer the client for download that is specific to their server then. Got it.

**Comment by ZopharShinta** [comment]
I believe this is handled in my pull request #595.

We'll need to decide what other functions we want expose to the Lua side.
Xp, Health, Stamina,, Enhancements, Temp powers, Clues, Contacts List, Waypoints, Inspirations, spawning NPCs and possibly others.

**Comment by broxen** [comment]
I would add to your list, new player spawn (with starting powers, insps, Inf, etc), and on levelup script.

**Comment by broxen** [comment]
Oh and SendMessage (text in chat window) #148

**Comment by ZopharShinta** [comment]
My commit from 10-2-18 for #595 added this functions to Lua.

Character.giveXp(client, int xp
Character.giveDebt(client, int debt)
Character.giveHp(client, float hp)
Character.giveEnd(client, float end)
Character.giveInf(client, int inf)
Character.giveInsp(client, string inspirationName)
Character.giveEnhancement(client, string enhancementName)

**Comment by ZopharShinta** [comment]
Ok I have the following added

- floating info
- floating damage
- targeting  - This is a response to the player targeting someone. Not sure if you're wanting to force the player to target someone.
- face location
- face entity
- sendMessage 

This leaves

- Temp powers
- Clues  - Adding them to the list, once we have a list.
- Contacts/ Contact List
- Waypoints
- Spawning NPCs
- new player spawn (with starting powers, insps, Inf, etc) - Not entirely sure what you're thinking here.
- levelup script
- adding/removing/setting tasks
- awarding souvenirs

Does that sound right?
I'll have to see which ones I can make available now. Most will be waiting on server side additions.

**Comment by broxen** [comment]
This was implemented in #611 in v6.

Let's start a new [Meta] issue for tracking any new features for the Lua Scripting Engine.

Thanks Zophar, you rock!
